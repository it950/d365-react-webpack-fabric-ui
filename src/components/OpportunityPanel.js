import React, { Component } from 'react'
import { PrimaryButton, DefaultButton } from 'office-ui-fabric-react/lib/Button'
import { ChoiceGroup } from 'office-ui-fabric-react/lib/ChoiceGroup'
import { Panel, PanelType } from 'office-ui-fabric-react/lib/Panel'

class OpportunityPanel extends Component {
  render() {
    const { isOpen, onClosePanel } = this.props

    return (
      <div>
        <Panel
          isOpen={isOpen}
          type={PanelType.smallFixedFar}
          onDismiss={onClosePanel}
          headerText="Opportunity Panel"
          closeButtonAriaLabel="Close"
          onRenderFooterContent={() => {
            return (
              <div>
                <PrimaryButton
                  onClick={onClosePanel}
                  style={{ marginRight: '8px' }}
                >
                  Save
                </PrimaryButton>
                <DefaultButton onClick={onClosePanel}>Cancel</DefaultButton>
              </div>
            )
          }}
        >
          <ChoiceGroup
            options={[
              {
                key: 'A',
                text: 'Option A',
              },
              {
                key: 'B',
                text: 'Option B',
                checked: true,
              },
              {
                key: 'C',
                text: 'Option C',
                disabled: true,
              },
              {
                key: 'D',
                text: 'Option D',
                checked: true,
                disabled: true,
              },
            ]}
            label="Pick one"
            required={true}
          />
        </Panel>
      </div>
    )
  }
}

export default OpportunityPanel
