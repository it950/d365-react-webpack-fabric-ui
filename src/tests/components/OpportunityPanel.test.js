import React from 'react'
import { shallow } from 'enzyme'

import OpportunityPanel from '../../components/OpportunityPanel'

describe('OpportunityPanel', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<OpportunityPanel />)
    expect(wrapper).toMatchSnapshot()
  })
})
