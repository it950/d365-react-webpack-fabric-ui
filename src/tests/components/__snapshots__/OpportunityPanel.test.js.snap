// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpportunityPanel should render correctly 1`] = `
<div>
  <Panel
    closeButtonAriaLabel="Close"
    hasCloseButton={true}
    headerText="Opportunity Panel"
    isBlocking={true}
    isOpen={false}
    onRenderFooterContent={[Function]}
    type={1}
  >
    <ChoiceGroup
      label="Pick one"
      options={
        Array [
          Object {
            "key": "A",
            "text": "Option A",
          },
          Object {
            "checked": true,
            "key": "B",
            "text": "Option B",
          },
          Object {
            "disabled": true,
            "key": "C",
            "text": "Option C",
          },
          Object {
            "checked": true,
            "disabled": true,
            "key": "D",
            "text": "Option D",
          },
        ]
      }
      required={true}
    />
  </Panel>
</div>
`;
